guidance/include/guidance/guidance_worker.hpp

```diff
@@ -90,7 +90,7 @@ namespace guidance
         private:
             GuidanceStateMachineFactory guidance_state_machine_factory;
             // Guidance state machine
-            GuidanceStateMachine* gsm;
+            std::auto_ptr<GuidanceStateMachine> gsm;
```

Comment:
don't use std::auto_ptr. In c++ 11 that is a deprecated class. Instead you should use std::shared_ptr, std::unique_ptr, or std::weak_ptr depending on the use case. I think std::unique_ptr is probably sufficient here, but if you are passing gsm into functions then std::shared_ptr might be better.